package/files/update_rmt_app_dir_permissions.sh

```diff
@@ -8,11 +8,26 @@ IFS=" "
 
 app_dir=$1
 app_dir_ownership=$(stat -c "%U %G" $app_dir)
+skip_public_tools="tools"
 
 if [[ $app_dir_ownership == "_rmt nginx" ]]; then
   # Sort application directory ordered by directory depth to
   # ensure secure recursive ownership change.
-  find -P $app_dir -type d | ruby -e 'dirs=readlines; dirs.each { |dir| puts("#{dir.strip} #{dir.strip.length}") }' | sort -k 2 -n | awk '/ / {print $1}' | xargs -I {} chown -h root:root {}
+  find -P $app_dir -type d ! -name $skip_public_tools | ruby -e 'dirs=readlines; dirs.each { |dir| puts("#{dir.strip} #{dir.strip.length}") }' | sort -k 2 -n | awk '/ / {print $1}' | xargs -I {} chown -h root:root {}
```

Comment:
why do we skip the tools directory here? We discussed the secrets, but I don't know why this would be necessary

Reply:
and why does it need to be owned by the nginx user? Can't it be owned by the root users and be world readable?

Reply:
Ah okay, that's possible too. But the entire `public/` sub-directory is nginx readable. Is there a good reason to perform your suggestion?

Reply:
readable isn't a problem. The service user shouldn't own any files that it doesn't need to own